x/m3em/node/types.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3db
+
+import (
+	"github.com/m3db/m3em/node"
+
+	"github.com/m3db/m3x/instrument"
+)
+
+// Node represents a ServiceNode pointing to an M3DB process
+type Node interface {
+	node.ServiceNode
+
+	// Health returns the health for this node
+	Health() (NodeHealth, error)
+
+	// Bootstrapped returns whether the node is bootstrapped
+	Bootstrapped() bool
+
+	// TODO(prateek): add more m3db service endpoints in ServiceNode
+	// - query service observable properties (nowFn, detailed_status)
+	// - set nowFn offset
+	// - logs
+	// - metrics
```

Comment:
Perhaps just remove this TODO for now, I think this is good for now and we can add methods as we see necessary.